src/datarobotx/idp/batch_predictions.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2024 DataRobot, Inc. and its affiliates.
+#
+# All rights reserved.
+#
+# DataRobot, Inc.
+#
+# This is proprietary source code of DataRobot, Inc. and its
+# affiliates.
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+# https://www.datarobot.com/wp-content/uploads/2021/07/DataRobot-Tool-and-Utility-Agreement.pdf
+
+from typing import Optional, TypedDict
+
+import datarobot as dr
+from datarobot.models.batch_job import IntakeSettings, OutputSettings, Schedule
+from datarobot.models.batch_prediction_job import BatchPredictionJobDefinition
+
+
+class JobSpec(TypedDict):
```

Comment:
Is this a straight copy from the python SDK somewhere or did you pick and choose a the essential items from something like https://github.com/datarobot/public_api_client/blob/2ca5850e82c31016c5694478f7087f0f229849df/datarobot/models/batch_prediction_job.py#L415?  Deferring to @brau0300 if we should be defining a new class rather than expecting the user to pass in a config dictionary since there are some additional parameters a user could pass in